lib/model/avatar_url.dart

```diff
@@ -0,0 +1,148 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:path/path.dart';
+
+import '../log.dart';
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// DEFAULT_AVATAR_SIZE in zerver/lib/upload.py.
+const int defaultUploadSizePx = 100;
+
+/// MEDIUM_AVATAR_SIZE in zerver/lib/upload.py.
+const int mediumUploadSizePx = 500;
+
+abstract class AvatarURL {
+  static AvatarURL fromUserOrBotData({
```

Comment:
nit:
```suggestion
  static AvatarURL fromUserData({
```

The "or bot" thing in zulip-mobile corresponds to the `UserOrBot` type there, which in turn is a legacy of some quirks in the Zulip API that zulip-mobile only partly papers over and zulip-flutter does so more thoroughly. In zulip-flutter, the corresponding type is just `User`.

Reply:
[fa5e14b](https://github.com/zulip/zulip-flutter/pull/534/commits/fa5e14b9a1d5c49886f9bfaccb1210b9d10e56cd)